File: 프로그래머스/프로그래머스_Lv1/가장 많이 받은 선물/[PCCP 기출문제] 이웃한 칸/이웃한칸.java
class Solution {
    
    private static final int[][] arr = {{-1, 0}, {1, 0}, {0, 1}, {0, -1}};
    
    public int solution(String[][] board, int h, int w) {
        String target = board[h][w];
        
        int count = 0;
        
        for(int[] each : arr) {
            int newH = h + each[0];
            int newW = w + each[1];
            
            if(newH < 0 || newH >= board.length || newW < 0 || newW >= board[0].length) continue;
            if(target.equals(board[newH][newW])) count++;
        }
        
        return count;
    }
}

File: 프로그래머스/프로그래머스_Lv1/가장 많이 받은 선물/[PCCP 기출문제] 이웃한 칸/이웃한칸.js
const array = [
  [-1, 0],
  [1, 0],
  [0, 1],
  [0, -1],
];

const solution = (board, h, w) => {
  let count = 0;

  for (each of array) {
    const newH = h + each[0];
    const newW = w + each[1];

    if (
      newH < 0 ||
      newH >= board.length() ||
      newW < 0 ||
      newW >= board[0].length
    )
      continue;
    if (board[newH][newW] === board[h][w]) count++;
  }

  return count;
};
